java/com/google/flatbuffers/FlatBufferBuilder.java

```diff
@@ -106,6 +106,20 @@ public FlatBufferBuilder init(ByteBuffer existing_bb){
         return this;
     }
 
+    /**
+     * Reset the FlatBufferBuilder by purging all data that it holds.
+     */
+    public void clear(){
+        space = bb.capacity();
+        bb.clear();
+        minalign = 1;
+        while(vtable_in_use > 0) vtable[--vtable_in_use] = 0;
+        vtable_in_use = 0;
```

Comment:
please double-check the list of instance variables, as I think you're missing some of them, like `nested` etc.